contracts/PBAB+Collabs/iyk/GenArt721CoreV2_IYK.sol

```diff
@@ -0,0 +1,195 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "../GenArt721CoreV2_PBAB.sol";
+
+import "@openzeppelin-4.5/contracts/utils/cryptography/ECDSA.sol";
+import "./ISignVerifierRegistry.sol";
+
+pragma solidity 0.8.9;
+
+/**
+ * @title Powered by Art Blocks ERC-721 core contract.
+ * Support for IYK pull mechanism to transfer tokens based on verified signatures
+ * @author Art Blocks Inc.
+ */
+contract GenArt721CoreV2_IYK is GenArt721CoreV2_PBAB {
+    using ECDSA for bytes32;
+
+    // Address of registry that resolves the signVerifier
+    ISignVerifierRegistry public signVerifierRegistry;
+    bytes32 public signVerifierId;
+
+    // Nonce per user, used to prevent replay signature attacks
+    mapping(address => uint256) public claimNonces;
```

Comment:
Nit/curious: why make this public (rather than private) if also exposing an external `getClaimNonce`?

Reply:
Good q! It's to maintain API parity with our older contracts that have this as `public` instead of `getClaimNonce`.

Will do a quick scan here too since this might not need to be `public` anymore cc @clee681

Reply:
Roger that–SGTM; may be worth clarifying in a `@dev` comment, but this is more of a nit–will defer to y'all.